client/lib/post-normalizer/rule-content-detect-media.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * External Dependencies
+ */
+import { map, compact, includes, some } from 'lodash';
+
+/**
+ * Internal Dependencies
+ */
+import { iframeIsWhitelisted } from './utils';
+
+/** Checks whether or not an image is a tracking pixel
+* @param {Node} image - DOM node for an img
+* @returns {boolean} isTrackingPixel - returns true if image is probably a tracking pixel
+*/
+function isTrackingPixel( image ) {
+	if ( ! image || ! image.src ) {
+		return false;
+	}
+
+	const edgeLength = image.height + image.width;
+	return edgeLength === 1 || edgeLength === 2;
+}
+
+function isCandidateForContentImage( image ) {
+	if ( ! image || ! image.getAttribute( 'src' ) ) {
+		return false;
+	}
+
+	const ineligibleCandidateUrlParts = [
+		'gravatar.com',
+		'/wpcom-smileys/',
+	];
+
+	const imageUrl = image.getAttribute( 'src' );
+
+	const imageShouldBeExcludedFromCandidacy = some( ineligibleCandidateUrlParts,
+		( urlPart ) => includes( imageUrl.toLowerCase(), urlPart )
+	);
+
+	return ! ( isTrackingPixel( image ) || imageShouldBeExcludedFromCandidacy );
+}
+
+/** detects and returns  metadata if it should be considered as a content image
+* @param {image} image - the image
+* @returns {object} metadata - regarding the image or null
+*/
+const detectImage = ( image ) => {
+	if ( isCandidateForContentImage( image ) ) {
+		return {
+			src: image.getAttribute( 'src' ),
+			width: image.width,
+			height: image.height,
+			mediaType: 'image',
+		};
+	}
+	return false;
+};
+
+// get the iframe src that autoplays the embed if we know how to, else null
+const getAutoplayIframe = ( iframe ) => {
+	if ( iframe.src.indexOf( 'youtube' ) > 0 ) {
+		const autoplayIframe = iframe.cloneNode();
+		if ( autoplayIframe.src.indexOf( '?' ) === -1 ) {
+			autoplayIframe.src += '?autoplay=1';
+		} else {
+			autoplayIframe.src += '&autoplay=1';
+		}
+		return autoplayIframe.outerHTML;
+	}
+	return null;
+};
+
+// get a picture thumnail version of the embed if it exists, else null
+const getThumbnailUrl = ( iframe ) => {
+	if ( iframe.src.indexOf( 'youtube' ) > 0 ) {
+		// grabbed from: http://stackoverflow.com/questions/3452546/javascript-regex-how-to-get-youtube-video-id-from-url
+		const regExp = /^.*(youtu\.be\/|v\/|u\/\w\/|embed\/|watch\?v=|\&v=)([^#\&\?]*).*/;
```

Comment:
I like the idea! 
I'll look around for a well-supported package -- that one only has 5 stars.